cc1120_arduino_tests/cc1120_spi.cpp

```diff
@@ -0,0 +1,444 @@
+ * 
+ * @param data - The array of data to write to the FIFO.
+ * @param len - The number of registers to write.
+ * @return true - If the write was successful.
+ * @return false - If the status byte is invalid.
+ */
+bool arduinoBurstWriteFIFO(uint8_t data[], uint8_t len) {
+    digitalWrite(CS, LOW);
+    if (!sendByteReceiveStatus(BURST_BIT | CC1120_REGS_FIFO_ACCESS_STD)) {
+        digitalWrite(CS, HIGH);
+        return false;
+    }
+    for(uint8_t i = 0; i < len; i++) {
+        SPI.transfer(data[i]);
+    }
+    digitalWrite(CS, HIGH);
+    return true;
+}
+
+/**
+ * @brief Reads directly from the FIFO on the CC1120.
+ * 
+ * @param addr - The address of the register to read. Range 0x00 - 0xFF.
+ * @param data - Pointer to the variable to store the read data.
+ * @return true - If the read was successful.
+ * @return false - If the register is not valid, or the status byte is invalid.
+ */
+bool arduinoReadFIFODirect(uint8_t addr, uint8_t *data) {
+    if (addr < CC1120_REGS_FIFO_TX_START || addr > CC1120_REGS_FIFO_RX_END) {
+        Serial.println("Not a valid FIFO register!");
+        return false;
+    }
+
+    digitalWrite(CS, LOW);
+    if (!sendByteReceiveStatus(CC1120_REGS_FIFO_ACCESS_DIR)) {
+        digitalWrite(CS, HIGH);
+        return false;
+    }
+    SPI.transfer(addr);
+    *data = SPI.transfer(0xff);
+    digitalWrite(CS, HIGH);
+    return true;
+}
+
+/**
+ * @brief - Reads consecutive registers directly from the FIFO on the CC1120 in burst mode.
+ * 
+ * @param addr - The address of the first register to read. Range 0x00 - 0xFF.
+ * @param data - The array to store the read data.
+ * @param len - The number of registers to read.
+ * @return true - If the read was successful.
+ * @return false - If the register is not valid, or the status byte is invalid.
+ */
+bool arduinoBurstReadFIFODirect(uint8_t addr, uint8_t data[], uint8_t len) {
+    if (addr < CC1120_REGS_FIFO_TX_START || addr > CC1120_REGS_FIFO_RX_END) {
+        Serial.println("Not a valid FIFO register!");
+        return false;
+    }
+
+    digitalWrite(CS, LOW);
+    if (!sendByteReceiveStatus(CC1120_REGS_FIFO_ACCESS_DIR)) {
+        digitalWrite(CS, HIGH);
+        return false;
+    }
+    SPI.transfer(addr);
+    for(uint8_t i = 0; i < len; i++) {
+        data[i] = SPI.transfer(0xff);
+    }
+    digitalWrite(CS, HIGH);
+    return true;
+}
+
+/**
+ * @brief Writes directly to the FIFO on the CC1120.
+ * 
+ * @param addr - The address of the register to write to. Range 0x00 - 0xFF.
+ * @param data - The data to write to the register.
+ * @return true - If the write was successful.
+ * @return false - If the register is not valid, or the status byte is invalid.
+ */
+bool arduinoWriteFIFODirect(uint8_t addr, uint8_t data) {
+    if (addr < CC1120_REGS_FIFO_TX_START || addr > CC1120_REGS_FIFO_RX_END) {
+        Serial.println("Not a valid FIFO register!");
+        return false;
+    }
+
```

Comment:
The FIFO on the CC1120 holds 128 bytes max. We should check if the length of the data exceeds that.